### PR TITLE
190/Add function call via FourByteDirectory in ViewTransactionActivity

### LINKED ISSUE #190: Display function calls in TransactionViewActivity
resolve function-names via 4byte.directory

### PR DESCRIPTION
**Pull request by friedger**
This PR adds a text field for transactions that might be a function call to a contract.

If the service 4byte.directory can resolve the hex signature of the function call the text signatures are shown, if not a dash "-" is shown.

Fixes #190 

screenshot_1519420936

### CODE REVIEW

**Comment by friedger** [comment]
@ligi I don't fully understand 3. Do you want these files to be copied into the asset folder?

**Comment by ligi** [comment]
No - I would say it can use the filebasedstore as cache with the path in the cache directory. Can fetch it from network, but just once

**Comment by friedger** [comment]
I have added all function signatures now and added a cache. I have updated the description of this PR.

File: app/src/main/java/org/walleth/data/AppDatabase.java [hunk 1]
```diff
@@ -12,7 +12,7 @@
 import org.walleth.data.transactions.TransactionDAO;
 import org.walleth.data.transactions.TransactionEntity;
 
-@Database(entities = {AddressBookEntry.class, Token.class, Balance.class, TransactionEntity.class}, version = 1)
+@Database(entities = {AddressBookEntry.class, Token.class, Balance.class, TransactionEntity.class}, version = 2)
```
Comment (ligi) [inline app/src/main/java/org/walleth/data/AppDatabase.java]: why do we go to v2 here?

Comment (friedger) [reply app/src/main/java/org/walleth/data/AppDatabase.java]: Oh, sorry. Thanks for spotting. I was to lazy to restore all my accounts on my device after doing an upgrade on a different branch. (#2 to the rescue :-)

**Comment by ligi** [comment]
:+1: I reworked it locally a bit - change now on master

**Pull request closed without merging**